### DISCUSSION TITLE
`_hover` property should overwrite `style` property

### DISCUSSION THREAD

**Discussion by jackherizsmith** [chakra-ui/chakra-ui#8776] [Ideas]
### Description

I'm setting the `style` of HTML elements, so that I can dynamically use preset colours from our design system. For example, my `button` looks like:

```tsx
<Button  style={{...colour.action.default}} />
```

where `colour.action.default` is `{background: 'blue'}`, let's say.

The problem is the `_hover` property, for example where `colour.action.hover` is `{background: 'red'}`, doesn't override this:

```tsx
<Button style={{...colour.action.default}} _hover={{...colour.action.hover}} /> /* no change on hover */
```

I think clearly, `_hover` (and similar properties) should override the `style` property, since they respond to user actions.

_____

An alternative solution works...

```tsx
<Button {...colour.action.default} _hover={{...colour.action.hover}} />
```

...however TypeScript isn't happy:

```console
Type ...CSSProperties | undefined is not assignable to type ReactNode | undefined
```

If anyone out there can help me resolve that instead, it would be really helpful. However as I say, I do think the `_` properties that ChakraUI makes available, should overwrite any overlapping styles, including those set by `style`.

### Link to Reproduction

https://codesandbox.io/p/devbox/button-hover-lsl9vw

### Steps to reproduce

1. Set background colour of an element using `style`
2. Set the hover background of that element using `_hover`
3. Hover over the element to see it not working

### Chakra UI Version

chakra-ui/next-js: ^2.2.0, chakra-ui/react: ^2.8.2

### Browser

Google Chrome Version 127.0.6533.122

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by jackherizsmith** [comment]
I'm dumb. I'm a dum dum.

I've refactored the codebase to use `theme`, obviously, and that now works great, obviously.

I'm inclined to leave this ticket open since I do still think `_hover` styles should overwrite `style` styles. However I do not think that anyone should attempt to build a website like I just did.

**Comment by segunadebayo** [comment]
Inline `style` always wins over `_hover`. That is CSS, not a Chakra bug.

Put colors in the theme or on style props. You already landed on that.
